src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -604,6 +604,9 @@ trait WriteService {
     def partialArticleFieldsUpdate(articleToPartialPublish: domain.Article,
                                    articleFieldsToUpdate: Seq[PartialArticleFields.Value],
                                    language: String): PartialPublishArticle = {
+      val newAvailability = if (articleFieldsToUpdate.contains(PartialArticleFields.availability)) {
+        Some(articleToPartialPublish.availability)
+      } else None
```

Comment:
Jeg har litt lyst til at disse skal være en pattern matching slik at vi får en warning dersom noen legger til en verdi i enumen, men ikke håndterer den.

Er ikke veldig farlig, og jeg vet ikke om warningen er verdt mindre lesbarhet for alle, så du kan velge selv om du vil gjøre det.
MEN om du har lyst så kan det løses sånn:

```scala
def partialArticleFieldsUpdate(articleToPartialPublish: domain.Article,
                               articleFieldsToUpdate: Seq[PartialArticleFields.Value],
                               language: String): PartialPublishArticle = {

  val initialPartial = PartialPublishArticle(None, None, None, None, None)
  articleFieldsToUpdate.distinct.foldLeft(initialPartial)((partialPublishArticle, field) => {
    field match {
      case PartialArticleFields.availability =>
        partialPublishArticle.copy(availability = Some(articleToPartialPublish.availability))
      case PartialArticleFields.grepCodes =>
        partialPublishArticle.copy(grepCodes = Some(articleToPartialPublish.grepCodes))
      case PartialArticleFields.license =>
        partialPublishArticle.copy(license = articleToPartialPublish.copyright.flatMap(c => c.license))
      case PartialArticleFields.metaDescription if (language == Language.AllLanguages) =>
        partialPublishArticle.copy(metaDescription = Some(articleToPartialPublish.metaDescription))
      case PartialArticleFields.metaDescription =>
        partialPublishArticle.copy(metaDescription = Some(articleToPartialPublish.metaDescription.find(m => m.language == language).toSeq))
      case PartialArticleFields.tags if (language == Language.AllLanguages) =>
        partialPublishArticle.copy(tags = Some(articleToPartialPublish.tags))
      case PartialArticleFields.tags =>
        partialPublishArticle.copy(tags = Some(articleToPartialPublish.tags.find(t => t.language == language).toSeq))
    }
  })

}
```